src/Monolog/Logger.php

```diff
@@ -148,9 +148,9 @@ class Logger implements LoggerInterface, ResettableInterface
     protected $exceptionHandler;
 
     /**
-     * @var int Keeps track of depth to prevent infinite logging loops
+     * @var LogDepthInterface Keeps track of depth to prevent infinite logging loops
      */
-    private $logDepth = 0;
+    protected $logDepth;
```

Comment:
why protected ?

Reply:
Because I will create a Logger which extends Monolog

And then rewrite logDepth

Reply:
Well, if you want to replace the tracking of the log depth, what you need is an interface for that object (to allow providing another implementation) and a way to inject a custom implementation.

But inheritance-based extension points are the worse ones both in term of maintenance (it is a lot harder to write a BC layer for a protected hooking point than for a clean extension point) and in term of flexibility (if 2 packages extend the Logger to hook into different places, you have to choose between them, you cannot combine their extensions)